airlock/views/request.py

```diff
@@ -287,13 +285,23 @@ def group_presenter(release_request, relpath, request):
     visibilities = release_request.get_writable_comment_visibilities_for_user(
         request.user
     )
+    # are we on the group page?
+    if len(relpath.parts) == 1:
+        inline = False
+    else:
+        inline = True
```

Comment:
it's semi-tempting to do a refactor to add some convenience methods to `UrlPath` for things like this, although from a quick skim of the codebase we've not hit the 3-4 repeated usages I'd want to see to make me think it's worthwhile. (I'm thinking `relpath.is_group()` for here and also an accessor-type method `relpath.group` in place if `relpath.parts[0]`.

Reply:
Well, that's a whole thing

There are times when a UrlPath means "path to file in a workspace" and times when it means "actual url" and times where it means "request url that will include group". cf `bll.get_request_file_from_output_path and bll.get_request_file_from_urlpath`.

Previously, I tried to split those into separate types: OutputPath, UrlPath, RequestUrlPath (which had a `.group` as you suggest and a `.output_path` w/o the group), and use the right types to help us catch these distintions.

But at the time, I couldn't make it work cleanly.  It's probably worth trying again, and definitely would be a good refactor.

Reply:
yeah, I think it would be a really valuable refactor, if we have time.

Reply:
put a ticket on the triage pile https://github.com/opensafely-core/airlock/issues/622